### ISSUE TITLE
[Feature Request] Include dependency items

### ISSUE THREAD

**Issue by KiraYamatoSD** [JensForstmann/factorio-make-everything-generator#2]
Great tool. Is it possible to include the dependency items down to the basic materials that are required by the assembling machines such as Iron or copper plates?

Output: Express transport belt (Blue belt)
Input: Fast transport belt (Red belt), Iron gear wheel

Fast transport belt (Red belt) <== Transport belt (Yellow Belt), Iron gear wheel

**Comment by JensForstmann** [comment]
Good idea, but to be honest I don't have a good idea how to design it properly.

Let's take the example, that I want to produce the following inserter:

- long-handed (red)
- fast (blue)
- filter (purple)
- stack (green)
- stack filter (white)

All materials needed to craft them:

- Inserter
- Iron gear wheel
- Electronic circuit
- Iron plate
- Iron ore
- Copper cable
- Copper plate
- Copper ore
- Advanced circuit
- Plastic bar
- Coal
- Petroleum gas
- Crude oil

The problem is, that not everything is craftable in an assembling machine. The way the tool works right now is to set the crafting machine in the settings (defaults to a assembling machine), And this will be used for all exported recipes. In the inserter example we would need assembling machines, smelters, chemical plants (or even mining drills and pumpjacks). It's also not so easy to know which machine is suitable for which recipe (not sure if that is extractable from the game via cheat command).

Second problem are materials where there are multiple recipes for, petroleum gas in this example: How to decided which one to use? Basic oil processing, Advanced oil procress, Light oil cracking or Coal liquefaction.

Third problem: Mods. The possibilities with mods are endless. Problem 1 and 2 will be much bigger if mods can come into play.

Any ideas or suggestions are welcome!
